netutils/nist.py

```diff
@@ -0,0 +1,130 @@
+"""Functions building NIST URLs from the os platform values."""
+import re
+import typing as t
+
+
+def get_nist_urls_juniper_junos(os_platform_data: t.Dict[str, t.Any], api_key: str) -> t.List[str]:
+    """Create a list of possible NIST Url strings for JuniperPlatform.
+
+    Args:
+        api_key: NIST-API-KEY - Request here https://nvd.nist.gov/developers/request-an-api-key
+
+    Returns:
+        List of NIST CPE URLs that may contain platform data.
+    """
+    nist_urls = []
+    base_url = f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey={api_key}&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos"""
+
+    url_args = {"base_url": base_url, **os_platform_data}
+    url_args["delim_six"] = ":*" * 6
+    url_args["delim_seven"] = ":*" * 7
+
+    if os_platform_data.get("isspecial"):
+        url_args["type"] = url_args.get("type").lower()  # type: ignore
+        # juniper:junos:12.1x47
```

Comment:
would this be more accurate?
 
```python
# e.g. base_ext = "juniper:junos:12.1x47"
```

Reply:
if so, do the same for all

Reply:
I'm not sure I understand what you're ask/request is.

The comments in those are just showing a version that that conditional is meant to match.

Reply:
match what? `base_ext`? if so, just be specific